cfgov/v1/models/caching.py

```diff
@@ -52,39 +52,80 @@ def get_auth(self):
             access_token=self.access_token
         )
 
-    def get_payload(self, obj):
+    def get_payload(self, obj, action):
         return {
-            'action': 'invalidate',
+            'action': action,
             'objects': [obj]
         }
 
+    def delete(self, url):
+        action = "delete"
+        resp = requests.post(
+            os.environ['AKAMAI_FAST_PURGE_URL'],
+            headers=self.headers,
+            data=json.dumps(self.get_payload(obj=url, action=action)),
+            auth=self.auth
+        )
+        logger.info(
+            u'Attempted to {action} cache for page {url}, '
+            'got back response {message}'.format(
+                action=action,
+                url=url,
+                message=resp.text
+            )
+        )
+        resp.raise_for_status()
```

Comment:
I don't think all of this needs to be repeated two times for single-urls and two times for "all".

You could still combine these into one, method, and then just have `purge(url)` call that method with `action="invalidate"`.

You could still combine the two single-url methods and the two "all" methods into one that takes "action" as an argument, but just retain the existing method signature for `purge()`, and have `purge()` call that other method.

Reply:
I have simplified Akamai cache code to post a request that will invalidate or delete purge